pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -436,6 +436,12 @@ func doUpdateVol(ctx *volumemgrContext, status *types.VolumeStatus) (bool, bool)
 		return false, true
 	}
 
+	if ctx.persistType == types.PersistZFS && !ctx.vaultProcessed {
```

Comment:
Why don't we also need this check for ext4/fscrypt? 

Also, if we don't want to process things until ctx.vaultProcessed then shouldn't we have some check for that in Run()? I think it already waits for the vault to be unlocked there.

Reply:
See comment below.